test/string-checker.js

```diff
@@ -207,6 +207,32 @@ describe('modules/string-checker', function() {
         });
     });
 
+    describe('esprima options', function() {
+        var customDescription = 'in no way a real error message';
+        var customEsprima = {
+            parse: function(code, options) {
+                var error = new Error();
+                error.description = options.foobar === 'qux' ? customDescription : 'should never happen';
+                error.lineNumber = 1;
+                error.column = 0;
+
+                throw error;
+            }
+        };
+
+        it('uses custom esprima options when set in the config', function() {
+            checker = new Checker({ esprima: customEsprima });
+            checker.registerDefaultRules();
+            checker.configure({ esprimaOptions: { foobar: 'qux' } });
+
+            var errors = checker.checkString('import { foo } from "bar";');
+            var error = errors.getErrorList()[0];
```

Comment:
Read my notes below. It tests that the correct options are passed into the parse function by spying on the parse function.